modules/users/server/controllers/users/users.authentication.server.controller.js

```diff
@@ -125,30 +154,89 @@ exports.signup = function(req, res) {
  * Signin after passport authentication
  */
 exports.signin = function(req, res, next) {
+
+  var statsObject = {
+    namespace: 'signin',
+    counts: {
+      count: 1
+    },
+    time: new Date(),
+    tags: {},
+    meta: {}
+  };
+
   passport.authenticate('local', function(err, user, info) {
 
     if (err || !user) {
-      return res.status(400).send(info);
+      // Log the failure to signin
+      log('error', 'User signin failed. #3tfgbg-1', {
+        reason: 'Wrong credentials',
+        error: err || null
+      });
+
+      // Send signin failure to stats servers
+      statsObject.tags.status = 'failed';
+      statsObject.tags.reason = 'wrong-credentials';
```

Comment:
In stathat, `tags.status` and `tags.reason` will currently lead to statistics with names:

- `signin.count`
- `signin.count.status.[failed|success]`
- `signin.count.reason.[wrong-credentials|suspended|other|local]`

The tags in names are _not_ multiplied like `signin.count.status.[status].reason.[reason]`.

I think having reason without status will lead to confusion (which reasons are given for success and which one's for failing). So i would suggest naming the reason something like `failed:wrong-credentials`, to give the status explicitly. Or whatever other format is preferred.